File: apps/api/app/schemas/story.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class StorySchema(BaseModel):
    id: UUID
    track_id: UUID
    author_user_id: UUID
    lead: str
    body: str
    like_count: int
    is_liked: bool = False
    created_at: datetime


class StoryCreateSchema(BaseModel):
    lead: str
    body: str


class StoryUpdateSchema(BaseModel):
    lead: str | None = None
    body: str | None = None

File: apps/mobile/lib/features/story/domain/story.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'story.freezed.dart';
part 'story.g.dart';

@freezed
class Story with _$Story {
  const factory Story({
    required String id,
    required String trackId,
    required String authorUserId,
    required String lead,
    required String body,
    @Default(0) int likeCount,
    required DateTime createdAt,
  }) = _Story;

  factory Story.fromJson(Map<String, dynamic> json) => _$StoryFromJson(json);
}
